### PR TITLE
[KYUUBI #4430] Lineage supports the `lateral view explode` statement

### LINKED ISSUE #4430: [FEATURE][Lineage] Support `Lateral view explode` statement
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

Test case:
```
test("test lateral view explode") {
  withTable("t1", "t2") { _ =>
    spark.sql("CREATE TABLE t1 (a string, b string, c string) USING hive")
    spark.sql("CREATE TABLE t2 (a string, b string, c string) USING hive")
    val ret0 = exectractLineage("insert into t1 select 1,2,a.action " +
        "from t2 lateral view explode(split(c,'\\},\\{')) a as action")
    assert(ret0 == Lineage(
      List("default.t2"),
      List("default.t1"),
      List(
        ("default.t1.a", Set()),
        ("default.t1.b", Set()),
        ("default.t1.c", Set("default.t2.c"))
      )))
  }
}
```

### Motivation

_No response_

### Describe the solution

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by iodone**
close #4430 
### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala [hunk 1]
```diff
@@ -349,6 +349,20 @@ trait LineageParser {
           joinColumnsLineage(parentColumnsLineage, childColumnsLineage)
         p.children.map(extractColumnsLineage(_, nextColumnsLineage)).reduce(mergeColumnsLineage)
 
+      case p: Generate =>
+        val childColumnsLineageWithId =
```
Comment (ulysses-you) [inline extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala:353]: `childColumnsLineageWithId` -> `generateColumnsLineageWithId`

File: extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala [hunk 2]
```diff
@@ -349,6 +349,20 @@ trait LineageParser {
...
+          ListMap(p.generatorOutput.map(attrRef => (attrRef.toAttribute, p.references)): _*)
+            .map { case (k, attrs) => (k.exprId, attrs) }
```
Comment (ulysses-you) [inline extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala:355]: shall we merge the second `map {...}` into the first `map` ?

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master
